### ISSUE TITLE
Checking for updates

### ISSUE THREAD

**Issue by eirnym** [telegramdesktop/tdesktop#4643]
Please, add checking for updates and updating. The popular method is to use Sparkle.framework to achieve this goal. 

### Expected behaviour
Tell us what should happen
A menu item "Check for updates" in OS X menu panel and an option in preferences to enable or disable them.

### Actual behaviour
Tell us what happens instead
I have to check them manually all the time. 

### Configuration
**Operating system:**
Mac OS X
**Version of Telegram Desktop:**
1.2.17 
**Used theme**:
any

<details><summary><b>Logs</b>:</summary>
Insert logs here (if necessary)

</details>

**Comment by john-preston** [comment]
@eirnym There is autoupdate already, you can press Check for updates in Settings.

It is available only in the version from https://desktop.telegram.org, not from App Store.

**Comment by stek29** [comment]
> Version of Telegram Desktop:
> 2.1.17

Where did you even get this? It's not TDesktop, and it's not new Swift native macOS app -- it's a really really old app which is no longer maintained.

You can get tdesktop at https://desktop.telegram.org, and you can get new macOS native app at https://macos.telegram.org.

**Comment by john-preston** [comment]
@stek29 I thought he meant 1.2.17, but just misspelled 🤔

**Comment by stek29** [comment]
@john-preston seems so indeed...

**Comment by eirnym** [comment]
I want to have a button to check them manually, without turning on autoupdates

**Comment by stek29** [comment]
@eirnym open settings and click on check updates

**Comment by eirnym** [comment]
@stek29 there's no such button. if you don't believe me, I can to make a full screenshot of my settings

![zrzut ekranu 2018-05-06 o 18 33 18](https://user-images.githubusercontent.com/485399/39675497-052c163a-515c-11e8-9ae2-30b45b1a6a73.png)

**Comment by eirnym** [comment]
@stek29 Please, send screenshot of your settings with this button

**Comment by eirnym** [comment]
@john-preston Is there possible to check updates manually on macOS?

**Comment by john-preston** [comment]
@eirnym What do you mean by "on macOS"? If the version from AppStore, then no, only through the AppStore app. If the version from https://desktop.telegram.org — go to Settings and press "Check for updates".

**Comment by stek29** [comment]
@eirnym Check update automatically, click on check for updates, uncheck Update automatically if you want to.

**Comment by eirnym** [comment]
@stek29 Just after I enable autoupdate option it runs a procedure to check the update and update it right away despite if user wants it or not.

I want check an update and optionally download and install the update.

**Comment by stek29** [comment]
@eirnym so you want "check for updates" to be available when autoupdate is disabled?

**Comment by eirnym** [comment]
Yes, I want just this button or a menu item (as the most common place for it)

**Comment by eirnym** [comment]
Is it too hard to implement this?

**Comment by john-preston** [comment]
@eirnym I don't see any reason to check for updates without updating the app. It has no information about the update like changelog or anything, just that the new version is available. And you have no information to make a decision about you wanting some particular new version or not. So I think that current way is enough. If you want a new version - you enable auto update, it downloads a new version and installs it. If you don't want to update right now - just disable autoupdate.

**Comment by eirnym** [comment]
I have control when I want to download a new version. Not every connection and time is good for autoupdate

**Comment by eirnym** [comment]
talking about release notes, you have them on Github, they could be the same for updates

**Comment by anohren** [comment]
I had a hard time figuring out whether this issue was 1. a feature request for an already existing feature, 2. a bug report for erroneously hiding the option to manually check for updates in the very situation where it becomes the most relevant, or 3. a feature request for having that button actually do what it says, i.e. "Check" for updates instead of checking *and* updating.

I'm going to assume it's the 2nd, since someone decided to close my report of that bug without explanation and only a link to here.

> @eirnym so you want "check for updates" to be available when autoupdate is disabled?

I know I do. That's the very situaion in which the option to manually checking for updates becomes relevant, which means hiding it would be a bug.

> I don't see any reason to check for updates without updating the app. It has no information about the update like changelog or anything, just that the new version is available. And you have no information to make a decision about you wanting some particular new version or not

This is a fair point, which would support the following ideal behavior in the absence of a changelog when autoupdate is disabled:

1. Literally _checking_ for updates is done automatically.
2. Choosing to download and install the update is then done through the click of a button.

> So I think that current way is enough. If you want a new version - you enable auto update, it downloads a new version and installs it. If you don't want to update right now - just disable autoupdate.

The current way is defective. It is apparent that people won't find the option to manually update when autoupdate is disabled. It has been demonstrated by the existence of this issue, and with a screenshot.

That the task is still techincally possible to perform doesn't change the fact that the current design forces people to do something they don't want to do -- in this case enabling autoupdates -- in order to complete it. It also forces them to figure out that two UI elements that should reasonably be orthogonal to each other are in fact dependent on each other.

**Comment by anohren** [comment]
> We therefore assume that the user has lost interest or resolved the problem on their own.

Since this is a bug report there is no problem that can be resolved by end users. It would therefore be much more reasonable to conclude that _developers_ have lost interest, if anyone — not the users. There is no reason to close.

**Comment by eirnym** [comment]
I didn't lost an interest, update scheme I and @anohren mentioned is implemented  a long list of open source software I know. 

Now I'll discuss interaction with a user more than checking and update mechanism itself.

Option 1: 
Pop-up window as it's implemented in Sparkle.Framework for macOS applications. 
Little bit noisy for a user, but works well.

Option 2: 
Firefox if autoupdate is disabled, checks and possibly installs a new update in "about" dialog

Option 3:
Google Chrome shows an icon on the top right corner if autoupdate is disabled or impossible for some reason. 

Current behaviour of Telegram.app is to continue be silent even if I left autoupdate on but I revoke write permissions from a app folder. 

My vote is for an option 3, but don't be silent and give a user a manual update option

**Comment by anohren** [comment]
> We therefore assume that the user has lost interest

No point mentioning it again; yes, let's say users have lost interest. At least one just did. Unsubscribing.

**Comment by eirnym** [comment]
Should I ping this ticket every 2 mounts with worthless comments?

**Comment by Aokromes** [comment]
> Should I ping this ticket every 2 mounts with worthless comments?

well, it was 6 months, also, slate bot helps to find still opened tickets already fixed.

**Comment by eirnym** [comment]
Ok, I don't see it implemented in current version, so how it was fixed? 

<img width="416" alt="Zrzut ekranu 2021-04-23 o 22 07 00" src="https://user-images.githubusercontent.com/485399/115924619-46787280-a480-11eb-8680-fd746a1f7e7b.png">

<img width="417" alt="Zrzut ekranu 2021-04-23 o 22 06 54" src="https://user-images.githubusercontent.com/485399/115924613-44161880-a480-11eb-954f-8ea2c243d77a.png">

**Comment by Aokromes** [comment]
> Ok, I don't see it implemented in current version, so how it was fixed?
> 
> <img alt="Zrzut ekranu 2021-04-23 o 22 07 00" width="416" src="https://user-images.githubusercontent.com/485399/115924619-46787280-a480-11eb-8680-fd746a1f7e7b.png">
> 
> <img alt="Zrzut ekranu 2021-04-23 o 22 06 54" width="417" src="https://user-images.githubusercontent.com/485399/115924613-44161880-a480-11eb-954f-8ea2c243d77a.png">

"slate bot helps to find still opened tickets already fixed." i don't means this issue is fixed.

**Comment by eirnym** [comment]
I see. I think it shows "no new comments" after some time, not what is "fixed".

**Comment by eirnym** [comment]
Even I'm able to make this fixed, there's a lot of reports, that it's hard to build on Linux or macOS because I need to have very specific set of libraries and very specific versions of them in very specific places. Because of this Telegram packages were removed from linux repos. 

Is there any guidance on that matter?

**Comment by Aokromes** [comment]
> I see. I think it shows "no new comments" after some time, not what is "fixed".

people comments if it's unfixed, or if it's fixed.

**Comment by ilya-fedin** [comment]
> Because of this Telegram packages were removed from linux repos.

AFAIK it still in all repos it was previously in

**Comment by ilya-fedin** [comment]
> I don't know why you think it's the same as "fixed".

I don't know why you think we think it's the same as "fixed"

**Comment by ilya-fedin** [comment]
> If you want to implement it, go ahead!

Not really. The fact feature requests exists doesn't mean it will be really implemented.

**Comment by Aokromes** [comment]
> this "stale" bot wrote why it marks issue as "stale". I don't know why you think it's the same as "fixed".
> 
> If you want to implement it, go ahead! I'll be happy with his feature request implemented.

... 
slate: waiting for answer from people on issue to know if it's already fixed or not.
we don't think slate means it's fixed.
with the help of slate bot we found some issues already fixed and closed, others remains opened.

**Comment by ilya-fedin** [comment]
In case of bug reports stalebot helps to find bugs that were disappeared on their own or fixed, but issues were forgotten
In case of feature requests stalebot helps to find feature requests no one really wants anymore

**Comment by tooomm** [comment]
>In case of feature requests, stalebot helps to **find feature requests no one really wants anymore**

>with the help of slate bot we found some issues already fixed and closed, others remains opened.

And many got also closed unfixed... because people are annoyed by the bot and the constant need to spam issues with senseless comments and useless notifications. There are issues with more bot and bump comments than actual content. 🙄 

Don't confuse that with people not demanding certain feature anymore, please.

**Comment by ilya-fedin** [comment]
Well if people don't answer then it means they don't really want a fix xD
If they want a soon fix they can contribute it themselves, otherwise they should wait and pong when bot asks to ping. ¯\_(ツ)_/¯

**Comment by ilya-fedin** [comment]
Without the bot there will again be >1k issues with a lot of duplicates and it would be really hard to find something valuable for anyone who wants to find something, again.

**Comment by eirnym** [comment]
There's a problem with notifications. Labels don't generate them. So "pong back" doesn't work even if I'd be subscribed to or be a member of the whole project
